src/core/lender/tranche/restricted_operator.sol

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2020 Centrifuge
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity >=0.4.24;
+
+import "./operator.sol";
+
+// RestrictedOperator restricts the allowance of users
+contract RestrictedOperator is Operator {
+    mapping (address => uint) maxCurrency;  // uint(-1) unlimited access by convention
```

Comment:
Given this behaves very similarly to erc20 allowances I would call it like that as well:
```suggestion
    mapping (address => uint) investAllowance;  // uint(-1) unlimited access by convention
```

Reply:
I don't like investAllowance because it is not specific enough you could express it in token or currency